internal/test/e2e/cluster_test.go

```diff
@@ -595,16 +595,28 @@ func TestUploadDownloadBasic(t *testing.T) {
 	contracts, err := cluster.Bus.Contracts(context.Background(), api.ContractsOpts{})
 	tt.OK(err)
 
-	// broadcast the revision for each contract and assert the revision height
-	// is 0.
+	// assert contracts haven't been revised
 	for _, c := range contracts {
 		if c.RevisionHeight != 0 {
-			t.Fatal("revision height should be 0")
+			t.Fatalf("contract %v 's revision height should be 0 but is %d", c.ID, c.RevisionHeight)
 		}
-		tt.OK(w.RHPBroadcast(context.Background(), c.ID))
 	}
 
-	// mine a block to get the revisions mined.
+	// broadcast the revision for each contract, we empty the pool first to
+	// ensure the revision aren't mined together with contract formations, this
+	// would not be considered a revision and the revision height would not be
+	// updated.
```

Comment:
This is odd. Shouldn't this work regardless?

Reply:
It's a technicality, if a block gets mined where we both form and revise a contract in the same block, it's not seen as a revised contract but the contract's initial state. Since this test performs an assertion where we use the contract's revision height this little hack I guess that drains the pool ensures every contract effectively gets revised and we can assert the revision height is no longer 0. I'm open to change this though but once I figured out what was happening I though it wasn't that big of a deal.

Reply:
I wouldn't call this a technicality considering that it's a very likely scenario in production. The subscription API should definitely update you multiple times. Once for the initial contract and once for every revision of that contract in a block. I'd discuss it with Luke if that's not already the case.

Reply:
I discussed it with Luke but I believe he mentioned it was an unlikely scenario in production, at least in v2. I'll pick this back up.